### PR TITLE
Target CI to use LBS

### PR DESCRIPTION
**Pull request by Mandur**
# PR for issue #749

## What is being addressed

We reactivate the E2E CI using the LBS.

### CODE REVIEW

File: .github/workflows/e2e-ci.yaml [hunk 1]
```diff
@@ -73,6 +73,62 @@ jobs:
...
+      with:
+         install: true
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+
+    - name: Build Docker LoRaWanNetworkServer images
+      run: |
+        echo ${{ secrets.ACR_CREDENTIALS }} | docker login "loramoduleintegrationtest.azurecr.io" --username  "loramoduleintegrationtest" --password-stdin
+        docker buildx bake --push LoRaWanNetworkServerarm32
+        # docker buildx bake --push LoraBasicsStationarm32v7
```
Comment (danigian) [inline .github/workflows/e2e-ci.yaml:120]: Can we add a comment here that specifies that this docker image is not rebuilt every time as it's painful to do so? (time-wise)

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/DownstreamSender.cs [hunk 2]
```diff
@@ -40,6 +40,7 @@ public async Task SendDownstreamAsync(DownlinkPktFwdMessage message)
             {
                 var region = await this.basicsStationConfigurationService.GetRegionAsync(message.StationEui, CancellationToken.None);
                 var payload = Message(message, region);
+                Logger.Log(payload, LogLevel.Debug);
```
Comment (danigian) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/DownstreamSender.cs:43]: Should we get rid of this?

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs [hunk 3]
```diff
@@ -19,18 +19,21 @@ public sealed class ModuleConnectionHost : IAsyncDisposable
...
             IClassCDeviceMessageSender defaultClassCDevicesMessageSender,
             ILoRaModuleClientFactory loRaModuleClientFactory,
-            ILoRaDeviceRegistry loRaDeviceRegistry)
+            ILoRaDeviceRegistry loRaDeviceRegistry,
+            LoRaDeviceAPIServiceBase loRaDeviceAPIService)
         {
             this.networkServerConfiguration = networkServerConfiguration ?? throw new ArgumentNullException(nameof(networkServerConfiguration));
             this.classCMessageSender = defaultClassCDevicesMessageSender ?? throw new ArgumentNullException(nameof(defaultClassCDevicesMessageSender));
             this.loRaDeviceRegistry = loRaDeviceRegistry ?? throw new ArgumentNullException(nameof(loRaDeviceRegistry));
+            this.loRaDeviceAPIService = loRaDeviceAPIService ?? throw new ArgumentNullException(nameof(loRaDeviceAPIService)); ;
```
Comment (danigian) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs:36]: extra ; at the end of this row

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DownlinkMessageBuilder.cs [hunk 4]
```diff
@@ -76,7 +76,9 @@ internal static class DownlinkMessageBuilder
             if (receiveWindow == Constants.ReceiveWindow2)
             {
                 lnsRxDelay = (ushort)timeWatcher.GetReceiveWindow2Delay(loRaDevice);
-                tmst = rxpk.Tmst + CalculateTime(timeWatcher.GetReceiveWindow2Delay(loRaDevice), loRaDevice.ReportedRXDelay);
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (danigian) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DownlinkMessageBuilder.cs:79]: I think we don't need anymore the warning disable

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DownlinkMessageBuilder.cs [hunk 5]
```diff
@@ -101,7 +103,9 @@ internal static class DownlinkMessageBuilder
                     return new DownlinkMessageBuilderResponse(null, false);
                 }
 
-                tmst = rxpk.Tmst + CalculateTime(timeWatcher.GetReceiveWindow1Delay(loRaDevice), loRaDevice.ReportedRXDelay);
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (danigian) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DownlinkMessageBuilder.cs:106]: I think we don't need anymore the warning disable

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs (hunk 3 above)
Comment (Mandur) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs:36]: How can this even compile?

Comment (danigian) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs:36]: You can put as many ; as you want, it just won't do anything

File: .github/workflows/e2e-ci.yaml (hunk 1 above)
Comment (Mandur) [reply .github/workflows/e2e-ci.yaml]: Done

**Review by danigian — APPROVED** [review]

**Pull request merged**